Updated to work with Python >3.6 + OTP authentication

ashwinpra opened the pull request:
Made the following changes:
- Removed unnecessary libraries from `requirements.txt`
- Bumped up other libraries
- Replaced deprecated `_attr_value_as_string` method of bs4
- Replaced `is` with `==` (It throws `SyntaxWarning` with some types of literals from Python >=3.8)
- Updated README

ashwinpra commented:
ERP recently introduced OTP authentication for login, so I have made required changes to the code for it to work
<img width="695" alt="image" src="https://github.com/metakgp/gyft/assets/62993493/1f27b4a9-18ae-4168-af6b-48b22440ccfa">

proffapt reviewed (approved):
LGTM. Tested locally, works.

ashwinpra commented:
- After the last update, the [iitkgp-erp-login](https://github.com/proffapt/iitkgp-erp-login-pypi/) package was made, so I incorporated that into the script
- This package already asks the user for their roll number, so I removed the corresponding command-line argument (`--user`)
- Updated dates for Spring 2023-24 semester
- Added 3 courses to `subjects.json`:    
    - AI61003: LINEAR ALGEBRA FOR AI AND ML
    - CS60029: RANDOMIZED ALGORITHM DESIGN
    - CS60075: NATURAL LANGUAGE PROCESSING

proffapt reviewed (changes requested):
- I don't know how `gyft-serve` works, but integrating my package here will require a bit of extra work to be done on `gyft-serve` to get it working.
- Also date updates have a separate PR #87

proffapt commented on requirements.txt:
```diff
@@ -1,24 +1,8 @@
+google_api_python_client==2.90.0
+httplib2==0.22.0
+icalendar==5.0.7
+iitkgp_erp_login==2.1.3
```
```suggestion
iitkgp_erp_login==2.2.0
```

proffapt commented on gyft.py:
```diff
@@ -5,21 +5,12 @@
-#### Parsing ends
+import iitkgp_erp_login.erp as erp
 
 ERP_HOMEPAGE_URL = 'https://erp.iitkgp.ac.in/IIT_ERP3/'
 ERP_LOGIN_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/auth.htm'
 ERP_SECRET_QUESTION_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getSecurityQues.htm'
+ERP_OTP_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getEmilOTP.htm' # blame ERP for the typo
```
```suggestion
```
The package already includes these, just check once whether they are used else where or not ( I don't see them )

ashwinpra replied:
You're right, they're not required anymore, I hadn't noticed it.

proffapt commented on subjects.json:
```diff
@@ -679,10 +682,10 @@
-    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
     "IM29006": "OPERATIONS RESEARCH LAB.",
     "IM29008": "WORK SYSTEM DESIGN LAB.",
+    "IM29202": "WORK SYSTEM DESIGN LABORATORY",
+    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
```
```suggestion
    "IM29202": "WORK SYSTEM DESIGN LABORATORY",
    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
```
This change distorts the sorted order of courses, please correct it.

proffapt replied:
Not needed

proffapt reviewed (changes requested):
After the requested changes, CLI will be good to go.

proffapt commented:
@chirag-ghosh This PR is good to be merged, works like a charm. With my package all the possible cases of login are handled (tested robustly). It was only aimed at CLI applications but I will later add support for web apps. Before that I will add a temporary fix on gyft-serve.
